synapse/storage/databases/main/roommember.py

```diff
@@ -670,6 +670,28 @@ async def get_users_who_share_room_with_user(
 
         return user_who_share_room
 
+    @cached(
+        max_entries=500000,
+        cache_context=True,
+        iterable=True,
+        prune_unread_entries=False,
```

Comment:
What's the rationale for setting opting out of the auto-prune behavior?

Reply:
No real rationale, should I remove for the default?

Reply:
> No real rationale, should I remove for the default?

We would prefer that caches are prune-able, yes.